src/main/java/seedu/address/model/application/Application.java

```diff
@@ -2,66 +2,54 @@
 
 import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
 
-import seedu.address.model.tag.Tag;
-
-import java.util.Collections;
-import java.util.HashSet;
 import java.util.Objects;
-import java.util.Set;
 
 /**
- * Represents an Application in the address book.
+ * Represents an Application in the Application book.
  * Guarantees: details are present and not null, field values are validated, immutable.
  */
 public class Application {
 
     // Identity fields
-    private final CompanyName companyName;
-    private final ContactNumber contactNumber;
+    private final Company company;
+    private final Contact contact;
     private final Email email;
-    private final PositionApplied positionApplied;
-
-    // Data fields
-    private final Set<Tag> tags = new HashSet<>();
+    private final Position position;
+    private final Date date;
 
     /**
      * Every field must be present and not null.
      */
-    public Application(CompanyName companyName, ContactNumber contactNumber, Email email, PositionApplied positionApplied, Set<Tag> tags) {
-        requireAllNonNull(companyName, contactNumber, email, positionApplied, tags);
-        this.companyName = companyName;
-        this.contactNumber = contactNumber;
+    public Application(Company company, Contact contact, Email email, Position position, Date date) {
+        requireAllNonNull(company, contact, email, position, date);
+        this.company = company;
+        this.contact = contact;
         this.email = email;
```

Comment:
Will add a tag for email in Parser and update UG accordingly.

Reply:
sorry, i dont quite get this, could you elaborate more?